content/actions/managing-issues-and-pull-requests/adding-labels-to-issues.md

```diff
@@ -46,7 +46,7 @@ In the tutorial, you will first make a workflow file that uses the [`andymckay/l
           issues: write{% endif %}
         steps:
           - name: Label issues
-            uses: andymckay/labeler@5c59dabdfd4dd5bd9c6e6d255b01b9d764af4414
+            uses: andymckay/labeler@master
```

Comment:
```suggestion
            uses: andymckay/labeler@e6c4322d0397f3240f0e7e30a33b5c5df2d39e90
```

Reply:
@ethomson When master updates again, does this SHA become invalidated, causing the same issue?

Reply:
It doesn't become invalidated, per se - you should still be able to `uses: andymckay/labeler@andymckay/labeler@5c59dabdfd4dd5bd9c6e6d255b01b9d764af4414`.

(I just tested it again and was successful with that - can you point me to your workflow where `andymckay/labeler@5c59dabdfd4dd5bd9c6e6d255b01b9d764af4414` fails?)

But yes, we should keep these up-to-date as they change over time as a good measure.

Reply:
@ethomson I was just playing around with actions when I stumbled upon this in the documentation.  I have a VERY basic action script that adds a label to new "issues".  Here is the code.  The line with "@master" would fail when I followed the recommended version number in the docs.

```
# This is a basic workflow to help you get started with Actions

name: CI

# Controls when the workflow will run
on:
  issues:
    types: [opened]

  # Allows you to run this workflow manually from the Actions tab
  workflow_dispatch:

# A workflow run is made up of one or more jobs that can run sequentially or in parallel
jobs:
  label_issues:
    runs-on: ubuntu-latest
    permissions:
      issues: write
    steps:
      - name: Label issues
        uses: andymckay/labeler@master
        with:
          add-labels: "action"
          # repo-token: ${{ secrets.GITHUB_TOKEN }}
```

Reply:
I dug up a full failure log if you need it:

> 2021-11-23T22:21:51.3048865Z Found online and idle hosted runner in the current repository's organization account that matches the required labels: 'ubuntu-latest'
> 2021-11-23T22:21:51.4250818Z Waiting for a Hosted runner in the 'organization' to pick this job...
> 2021-11-23T22:21:51.5508617Z Job is waiting for a hosted runner to come online.
> 2021-11-23T22:21:54.6558359Z Job is about to start running on the hosted runner: Hosted Agent (hosted)
> 2021-11-23T22:21:57.9278748Z Current runner version: '2.284.0'
> 2021-11-23T22:21:57.9306242Z ##[group]Operating System
> 2021-11-23T22:21:57.9307165Z Ubuntu
> 2021-11-23T22:21:57.9307617Z 20.04.3
> 2021-11-23T22:21:57.9308006Z LTS
> 2021-11-23T22:21:57.9308479Z ##[endgroup]
> 2021-11-23T22:21:57.9309061Z ##[group]Virtual Environment
> 2021-11-23T22:21:57.9309644Z Environment: ubuntu-20.04
> 2021-11-23T22:21:57.9310192Z Version: 20211114.1
> 2021-11-23T22:21:57.9311100Z Included Software: https://github.com/actions/virtual-environments/blob/ubuntu20/20211114.1/images/linux/Ubuntu2004-README.md
> 2021-11-23T22:21:57.9312374Z Image Release: https://github.com/actions/virtual-environments/releases/tag/ubuntu20%2F20211114.1
> 2021-11-23T22:21:57.9313240Z ##[endgroup]
> 2021-11-23T22:21:57.9313802Z ##[group]Virtual Environment Provisioner
> 2021-11-23T22:21:57.9314434Z 1.0.0.0-master-20211117-1
> 2021-11-23T22:21:57.9314915Z ##[endgroup]
> 2021-11-23T22:21:57.9316120Z ##[group]GITHUB_TOKEN Permissions
> 2021-11-23T22:21:57.9317260Z Issues: write
> 2021-11-23T22:21:57.9317783Z Metadata: read
> 2021-11-23T22:21:57.9318366Z ##[endgroup]
> 2021-11-23T22:21:57.9321168Z Prepare workflow directory
> 2021-11-23T22:21:57.9925358Z Prepare all required actions
> 2021-11-23T22:21:57.9934644Z Getting action download info
> 2021-11-23T22:21:58.2158478Z Download action repository 'andymckay/labeler@5c59dabdfd4dd5bd9c6e6d255b01b9d764af4414' (SHA:5c59dabdfd4dd5bd9c6e6d255b01b9d764af4414)
> 2021-11-23T22:22:02.0631784Z ##[group]Run andymckay/labeler@5c59dabdfd4dd5bd9c6e6d255b01b9d764af4414
> 2021-11-23T22:22:02.0632807Z with:
> 2021-11-23T22:22:02.0633210Z   add-labels: action
> 2021-11-23T22:22:02.0633657Z ##[endgroup]
> 2021-11-23T22:22:03.0276203Z RequestError [HttpError]: Bad credentials
> 2021-11-23T22:22:03.0282186Z     at /home/runner/work/_actions/andymckay/labeler/5c59dabdfd4dd5bd9c6e6d255b01b9d764af4414/node_modules/@octokit/request/dist-node/index.js:66:23
> 2021-11-23T22:22:03.0285352Z     at processTicksAndRejections (internal/process/task_queues.js:93:5)
> 2021-11-23T22:22:03.0288224Z     at async label (/home/runner/work/_actions/andymckay/labeler/5c59dabdfd4dd5bd9c6e6d255b01b9d764af4414/label.js:25:33) {
> 2021-11-23T22:22:03.0290739Z   name: 'HttpError',
> 2021-11-23T22:22:03.0291354Z   status: 401,
> 2021-11-23T22:22:03.0291870Z   headers: {
> 2021-11-23T22:22:03.0298277Z     'access-control-allow-origin': '*',
> 2021-11-23T22:22:03.0301081Z     'access-control-expose-headers': 'ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Resource, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, Deprecation, Sunset',
> 2021-11-23T22:22:03.0303426Z     connection: 'close',
> 2021-11-23T22:22:03.0304054Z     'content-length': '80',
> 2021-11-23T22:22:03.0304911Z     'content-security-policy': "default-src 'none'",
> 2021-11-23T22:22:03.0305881Z     'content-type': 'application/json; charset=utf-8',
> 2021-11-23T22:22:03.0306629Z     date: 'Tue, 23 Nov 2021 22:22:02 GMT',
> 2021-11-23T22:22:03.0307826Z     'referrer-policy': 'origin-when-cross-origin, strict-origin-when-cross-origin',
> 2021-11-23T22:22:03.0308959Z     server: 'GitHub.com',
> 2021-11-23T22:22:03.0309968Z     'strict-transport-security': 'max-age=31536000; includeSubdomains; preload',
> 2021-11-23T22:22:03.0311106Z     vary: 'Accept-Encoding, Accept, X-Requested-With',
> 2021-11-23T22:22:03.0312033Z     'x-content-type-options': 'nosniff',
> 2021-11-23T22:22:03.0312819Z     'x-frame-options': 'deny',
> 2021-11-23T22:22:03.0313621Z     'x-github-media-type': 'github.v3; format=json',
> 2021-11-23T22:22:03.0314542Z     'x-github-request-id': '07C1:1E32:CD4CC:1F8720:619D698A',
> 2021-11-23T22:22:03.0315299Z     'x-ratelimit-limit': '60',
> 2021-11-23T22:22:03.0316034Z     'x-ratelimit-remaining': '59',
> 2021-11-23T22:22:03.0316784Z     'x-ratelimit-reset': '1637709722',
> 2021-11-23T22:22:03.0317523Z     'x-ratelimit-resource': 'core',
> 2021-11-23T22:22:03.0318250Z     'x-ratelimit-used': '1',
> 2021-11-23T22:22:03.0318897Z     'x-xss-protection': '0'
> 2021-11-23T22:22:03.0319333Z   },
> 2021-11-23T22:22:03.0319947Z   request: {
> 2021-11-23T22:22:03.0320489Z     method: 'GET',
> 2021-11-23T22:22:03.0321518Z     url: 'https://api.github.com/repos/joegiglio/actions1/issues/2',
> 2021-11-23T22:22:03.0322388Z     headers: {
> 2021-11-23T22:22:03.0323382Z       accept: 'application/vnd.github.v3+json',
> 2021-11-23T22:22:03.0324408Z       'user-agent': 'octokit.js/16.28.9 Node.js/12.13.1 (Linux 5.11; x64)',
> 2021-11-23T22:22:03.0325281Z       authorization: 'token [REDACTED]'
> 2021-11-23T22:22:03.0325787Z     },
> 2021-11-23T22:22:03.0326388Z     request: { hook: [Function: bound bound register], validate: [Object] }
> 2021-11-23T22:22:03.0326993Z   },
> 2021-11-23T22:22:03.0327813Z   documentation_url: 'https://docs.github.com/rest'
> 2021-11-23T22:22:03.0328473Z }
> 2021-11-23T22:22:03.0362257Z Cleaning up orphan processes

Reply:
Thanks @joegiglio!  This is really helpful - it looks like `andymckay/labeler@5c59dabdfd4dd5bd9c6e6d255b01b9d764af4414` is able to be found and run, but this version of the action itself has a bug (possibly when working with private repositories?)